pkg/route/hostassignment/assignment_test.go

```diff
@@ -197,6 +216,29 @@ func TestHostWithWildcardPolicies(t *testing.T) {
 			allow:          false,
 			errs:           1,
 		},
+		{
+			name:           "key-added",
+			host:           "host",
+			expected:       "host",
+			oldHost:        "host",
+			tls:            &routev1.TLSConfig{Termination: routev1.TLSTerminationEdge, Key: "a"},
+			oldTLS:         &routev1.TLSConfig{Termination: routev1.TLSTerminationEdge},
+			wildcardPolicy: routev1.WildcardPolicyNone,
+			allow:          false,
+			errs:           1,
+		},
+		{
+			name:           "key-removed",
+			host:           "host",
+			expected:       "host",
+			oldHost:        "host",
+			tls:            &routev1.TLSConfig{Termination: routev1.TLSTerminationEdge},
+			oldTLS:         &routev1.TLSConfig{Termination: routev1.TLSTerminationEdge, Key: "b"},
+			wildcardPolicy: routev1.WildcardPolicyNone,
+			allow:          false,
+			// removing key info is allowed
+			errs: 0,
+		},
```

Comment:
Do we need these tests for the change you did or you just wanted to enhance the unit tests for other cases?

Reply:
These tests are unrelated to the changes I made related to externalCertificate; I just wanted to enhance different test scenarios.